log/value.go

```diff
@@ -39,6 +39,36 @@ func Timestamp(t func() time.Time) Valuer {
 	return func() interface{} { return t() }
 }
 
+// TimestampFormat returns a Valuer that produces a TimeFormat value from
+// layout and the time returned by t. Users will probably want to use
+// DefaultTimestamp or DefaultTimestampUTC.
+func TimestampFormat(t func() time.Time, layout string) Valuer {
```

Comment:
Using `now` makes sense to me, but I think `t` can also be fine, if it's explained a bit more didactically; I think I want the comment here to explain the "why" rather than the "what", e.g.

```go
// TimestampFormat can be used to create custom time formats for log timestamps.
// The t function will be invoked to get the time to format; unless you're doing
// something tricky, pass time.Now. The layout string is passed to Time.Format.
//
// Most users will want to use DefaultTimestamp or DefaultTimestampUTC, which
// are TimestampFormats that use the RFC3339Nano format.
```